Sources/Recycling/ListViews/Wishlist/Cell/WishListViewCell.swift

```diff
@@ -0,0 +1,221 @@
+            subtitleContainer.trailingAnchor.constraint(equalTo: adImageView.trailingAnchor),
+            subtitleContainer.topAnchor.constraint(equalTo: adImageView.bottomAnchor, constant: .mediumSpacing),
+            
+            titleLabel.leadingAnchor.constraint(equalTo: subtitleContainer.leadingAnchor),
+            titleLabel.trailingAnchor.constraint(equalTo: subtitleContainer.trailingAnchor),
+            titleLabel.topAnchor.constraint(equalTo: subtitleContainer.bottomAnchor),
+        ])
+    }
+    
+    public override func layoutSubviews() {
+        let roundedCornerLayer = CAShapeLayer()
+        roundedCornerLayer.bounds = rightImageDetail.frame
+        roundedCornerLayer.position = rightImageDetail.center
+        roundedCornerLayer.path = UIBezierPath(roundedRect: rightImageDetail.bounds, cornerRadius: WishListViewCell.cornerRadius).cgPath
+        rightImageDetail.layer.mask = roundedCornerLayer
+        
+        let roundedRightCornerLayer = CAShapeLayer()
+        roundedRightCornerLayer.bounds = leftImageDetail.frame
+        roundedRightCornerLayer.position = leftImageDetail.center
+        roundedRightCornerLayer.path = UIBezierPath(roundedRect: leftImageDetail.bounds, byRoundingCorners: [UIRectCorner.topRight],
+                                                    cornerRadii: CGSize(width: WishListViewCell.cornerRadius,
+                                                                        height: WishListViewCell.cornerRadius)).cgPath
+        leftImageDetail.layer.mask = roundedRightCornerLayer
+    }
+    
+    // MARK: - Superclass Overrides
+    
+    override public func prepareForReuse() {
+        adImageView.image = nil
+        leftImageDetail.text = ""
+        rightImageDetail.text = ""
+        leftSubtitleDetail.text = ""
+        rightSubtitleDetail.text = ""
+        titleLabel.text = ""
+        
+        if let model = model {
+            dataSource?.wishListViewCell(self, cancelLoadingImageForModel: model, imageWidth: adImageView.frame.size.width)
+        }
+    }
+    
+    // MARK: - Depedency injection
+    
+    /// The model contains data used to populate the view.
+    public var model: WishListViewModel? {
+        didSet {
+            guard let model = model else { return }
+            leftImageDetail.text = model.leftImageDetail
+            rightImageDetail.text = model.rightImageDetail
+            leftSubtitleDetail.text = model.leftSubtitleDetail
+            rightSubtitleDetail.text = model.rightSubtitleDetail
+            titleLabel.text = model.title
+            accessibilityLabel = model.accessibilityLabel
+        }
+    }
+    
+    // MARK: - Public
+    
+    /// Loads the image for the `model` if imagePath is set
+    public func loadImage() {
+        if let model = model {
+            loadImage(model: model)
+        }
+    }
+    
+    // MARK: - Private
+    
+    private func loadImage(model: WishListViewModel) {
```

Comment:
Why abstracting all the logic to the private method instead of doing it inside the public one?

Reply:
To keep it consistent with the `FavoriteListView` implementation.